notebooks/icrp107_dataset/icrp107_dataset.ipynb

```diff
@@ -437,26 +462,11 @@
     "    add[\"Half_life_readable\"] = readable_half_life(line[1], unit)\n",
     "    if add[\"Radionuclide\"][-1].isalpha():\n",
     "        add[\"Metastable_state\"] = add[\"Radionuclide\"][-1]\n",
-    "        # loop to check for different labelling of metastable states in NuBase as opposed to ICRP-107\n",
-    "        for meta in [\"m\",\"n\",\"p\",\"q\",\"r\"]:\n",
-    "            meta_name = name[:-1] + meta\n",
-    "            if meta_name in nubase_df.index:\n",
-    "                nubase_info = nubase_df.loc[meta_name]\n",
-    "                # account for slight differences in ICRP-107 and NuBase half-life data\n",
-    "                if isclose(add[\"Half_life_s\"], nubase_info[\"Half_life_s\"], rel_tol=.15) == True:\n",
-    "                    isomer_excitation_energy = nubase_info.Isomer_excitation_energy\n",
-    "                    break\n",
-    "                elif meta_name.endswith(\"r\"):\n",
-    "                    print(meta_name, add[\"Half_life_s\"], nubase_info[\"Half_life_s\"])\n",
-    "            if meta == \"r\":\n",
-    "                if any([issue_names in name for issue_names in [\"Ta-178m\",\"Y-84m\",\"Pm-154m\",\"Pr-134m\"]]):\n",
-    "                    # possible error in NuBase - Ta-178 and Ta-178m have swapped half-lives, same for others\n",
-    "                    isomer_excitation_energy = nubase_df.loc[name].Isomer_excitation_energy\n",
-    "                else:\n",
-    "                    print(\"No matching data present in NuBase for\", name)\n",
-    "        add[\"Atomic_mass\"] = ame_mass + isomer_excitation_energy\n",
-    "    else:\n",
-    "        add[\"Atomic_mass\"] = ame_mass\n",
+    "    ame_mass = ame2020_df.loc[name.strip(\"nm\")][\"Mass\"]\n",
+    "    # set mass to AME2020 data plus isomer excitation energy matched from nubase\n",
+    "    isomer_excitation_energy = excitation_energy(add[\"Element\"], add[\"A\"],\n",
+    "                                                 add[\"Half_life_s\"], nubase_df)\n",
+    "    add[\"Atomic_mass\"] = ame_mass + isomer_excitation_energy\n",
```

Comment:
@lemointm I like the way you have done this section with the new function call.

Looking at the diffs from the last commit in icrp.csv, I noticed that there were some more cases other than Ta-178/m, Y-84m, Pr-154/m & Pr-134m that were also fixed with the new logic. However there are also three new cases that break because of it! And all for different reasons...

- Te-123 - the half-life in ICRP-07 is 6.00E+14 y, but in Nubase2020 it is stable. The stable nuclides don't get added to the `nubase_df`, so what happens is `excitation_energy()` assigns the excitation energy for Te-123m (hl 119.2 days) as that is all it has to compare to.
- Ru-219 - ICRP-07 hl is 10 ms, Nubase2020 is 9 ms. However, in Nubase2020 there is an Ru-219m with a half-life of 10 ms, so it gets matched to this. However Nubase2020 says Ru-219m was discovered in 2018, so it post-dates the creation of ICRP-07. So the correct match should probably be Ru-219, so zero excitation energy.
- Pb-195m - ICRP-07 & Nubase2020 hl is 15 min, but it matches with Pb-195 which also has a half-life of 15 min! It should be picking up the non-zero excitation energy, like was the case with the old prior commit logic.

I ran checks locally and these three issues don't affect anything else, luckily. Writing general logic code to fix these issues will be more effort than it is worth TBH. This has already become a time sink. I recommend just including a manual override instead. If you pass the nuclide `name` as a parameter into `excitation_energy()`, then immediately before the return statement you could include code like:

```
overrides = {"Te-123": 0.0,
              "Ru-219": compare_df.loc["Ru-219"].Isomer_excitation_energy,
              "Pb-195m: compare_df.loc["Pb-195m"].Isomer_excitation_energy}
if name in overrides:
    isomer_excitation_energy = overrides[name]
```

to fix the problem. Also please add a comment referencing this PR or the reasons why overrides are needed.

Reply:
Sounds good @alexmalins! The last commit should have fixed these issues (I think you meant Ra-219, correct me if I'm wrong). Thank you for all the input, I really appreciate it!